### PR TITLE
Revamp MenuButtonBuilder

### PR DESCRIPTION
**Pull request by jpnurmi**
The old `MenuButtonBuilder` is no longer used in any of our apps so I repurposed it to something that uses the new overlay-based `MenuAnchor` to open the menu without any mobile-oriented transitions (see https://github.com/ubuntu/yaru_widgets.dart/issues/557).

### Before

[Screencast from 2023-02-19 18-20-16.webm](https://user-images.githubusercontent.com/140617/219964074-9f8c7498-d649-42a4-a0fd-a4f755a6415e.webm)

### After

[Screencast from 2023-02-19 18-22-06.webm](https://user-images.githubusercontent.com/140617/219964083-c26ebbfc-e5f8-41d4-935b-5152a6bb374c.webm)

### CODE REVIEW

**Comment by jpnurmi** [comment]
[Screencast from 2023-02-21 12-04-22.webm](https://user-images.githubusercontent.com/140617/220331668-9aeb4e16-584e-471a-9230-aa0e7e8039bc.webm)

File: packages/ubuntu_widgets/lib/src/menu_button_builder.dart [hunk 1]
```diff
@@ -16,62 +17,140 @@ import 'package:flutter/material.dart';
...
-            checked: value == selected,
-            child: itemBuilder(context, value, null),
-          );
-        }).toList();
+    final box = context.findRenderObject() as RenderBox?;
+    return MenuAnchor(
+      controller: _controller,
+      crossAxisUnconstrained: false,
+      style: MenuStyle(
+        minimumSize: MaterialStatePropertyAll(Size(box?.size.width ?? 0, 0)),
```
Comment (jpnurmi) [inline packages/ubuntu_widgets/lib/src/menu_button_builder.dart:85]: the width is off when rebuilt with a different selected value

**Review by d-loose — APPROVED** [review]
:+1:

**Comment by jpnurmi** [comment]
Oops, this is more work than I expected. :grin: It needs to handle focus as well...

**Comment by jpnurmi** [comment]
still some problems with the input decoration needed for this one

image

**Pull request merged**